distribution/docker/src/docker/Dockerfile

```diff
@@ -257,6 +259,21 @@ RUN sed -i -e 's/ES_DISTRIBUTION_TYPE=tar/ES_DISTRIBUTION_TYPE=docker/' bin/elas
     find . -xdev -perm -4000 -exec chmod ug-s {} + && \\
     find . -type f -exec chmod o+r {} +
 
+<% if (docker_base == "cloud") { %>
+# Preinstall common plugins
```

Comment:
Has there been a discussion around other plugins that are now part of the cloud image and customers can install ? 
AFAIK some of the popular examples are the `analysis-*` plugins.  We have options either way: bundle them in the image, but don't install them by default, or have cloud install them over the download service ( and document that it doesn't work for air gaped installation, these would install it like they install a custom plugin ). 7.x and ECE 2.x are think sensible to this approach we would need to seek PM approval as it would I think be a breaking change for such environments I'm not sure it's ok to make or not.

Reply:
I wasn't involved in the discussions to date, I was going by #74795. Once we have something work end-to-end for these plugins, from the ES perspective it ought to be straightforward to add more.

Reply:
We discussed this an concluded that for non-essential plugins (ones that aren't _always_ installed for any given cloud provider) they would simply be downloaded as with an on-prem install.

Reply:
After further discussion, this PR now builds a `cloud` image and a `cloud-ess` image. The former image installs only the 3 repository plugins, and a plugin tool wrapper that checks in an archive directory for other plugins before reaching out to the internet. The latter image includes all the other plugins in the archive directory, in a "batteries-included" manner.